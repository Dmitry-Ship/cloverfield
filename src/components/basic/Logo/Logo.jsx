import PropTypes from 'prop-types';
import React from 'react';
import { Link } from 'react-router-dom';
import { logo } from './Logo.scss';

const Logo = ({ className, ...rest }) => (
  <Link className={`${logo} ${className}`} {...rest} >
    Cloverfield
  </Link>
);

export default Logo;

Logo.defaultProps = {
  className: '',
  to: '/',
};

Logo.propTypes = {
  className: PropTypes.string,
  to: PropTypes.string,
};
