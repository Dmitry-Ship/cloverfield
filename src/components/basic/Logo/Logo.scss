@value colors: "../../../styles/colors.scss";
@value accent from colors;
@value breakPoints: "../../../styles/breakPoints.scss";
@value mobile from breakPoints;

.logo {
  color: accent;
  font-size: 2rem;
  font-weight: bold;
  font-family: Pacifico;
  @media mobile {
    font-size: 1.25rem;
  }
}

