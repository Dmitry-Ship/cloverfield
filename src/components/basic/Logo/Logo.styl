@value colors "../../../styles/colors.styl";
@value accent from colors;

.logo
	color accent
	font-size 2rem
	font-weight bold
	font-family Raleway
	+below(600px)
		font-size: 1.25rem
