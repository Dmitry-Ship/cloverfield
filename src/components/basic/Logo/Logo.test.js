import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Logo from './Logo';

describe('<Logo />', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <Logo />
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
