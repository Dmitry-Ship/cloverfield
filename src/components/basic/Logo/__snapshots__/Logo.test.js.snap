// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should render correctly 1`] = `
<a
  className="logo "
  href="/"
  onClick={[Function]}
>
  Cloverfield
</a>
`;
